pkg/transportrequest/uploadcts_test.go

```diff
@@ -0,0 +1,156 @@
+package transportrequest
+
+import (
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestUploadCTS(t *testing.T) {
+
+	filesMock := mock.FilesMock{}
+	files = &filesMock
+	defer func() { files = piperutils.Files{} }()
+
+	t.Run("npm install command tests", func(t *testing.T) {
+		cmd := mock.ShellMockRunner{}
+		action := CTSUploadAction{
+			Connection:  CTSConnection{Endpoint: "", Client: "", User: "me", Password: "******"},
+			Application: CTSApplication{Pack: "", Name: "", Desc: ""},
+			Node: CTSNode{
+				DeployDependencies: []string{"@sap/my-dep"},
+				InstallOpts:        []string{"--verbose", "--registry", "https://registry.example.org"},
+			},
+			TransportRequestID: "12345678",
+			ConfigFile:         "ui5-deploy.yaml",
+			DeployUser:         "node",
+		}
+		err := action.Perform(&cmd)
+		if assert.NoError(t, err) {
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"npm install --global --verbose --registry https://registry.example.org @sap/my-dep",
+			)
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"su node",
+			)
+		}
+	})
+
+	t.Run("deploy command tests", func(t *testing.T) {
+		t.Run("all possible values provided", func(t *testing.T) {
+			cmd := mock.ShellMockRunner{}
+			action := CTSUploadAction{
+				Connection:  CTSConnection{Endpoint: "https://example.org:8080/cts", Client: "001", User: "me", Password: "******"},
+				Application: CTSApplication{Pack: "abapPackage", Name: "appName", Desc: "the Desc"},
+				Node: CTSNode{
+					DeployDependencies: []string{},
+					InstallOpts:        []string{},
+				},
+				TransportRequestID: "12345678",
+				ConfigFile:         "ui5-deploy.yaml",
+				DeployUser:         "doesNotMatterInThisCase",
+			}
+
+			err := action.Perform(&cmd)
+			if assert.NoError(t, err) {
+				assert.Contains(
+					t,
+					cmd.Calls[0],
+					"fiori deploy -f -y --username ABAP_USER --password ABAP_PASSWORD -e \"the Desc\" --noConfig -u https://example.org:8080/cts -l 001 -t 12345678 -p abapPackage -n appName",
```

Comment:
Is it not necessary to have a dollar in front of the `ABAP_USER` and `ABAP_PASSWORD,` as they are passed from the env, are they not?

Reply:
> Is it not necessary to have a dollar in front

No. It should be like it is. We do not hand over the values for `ABAP_USER` and `ABAP_PASSWORD` here. We 'only' denote which environment variables are expected to hold the corresponding values. These values are retrieved later on inside the fiori upload toolset.

Reply:
Ok, got it.